app/src/main/java/com/isc/hermes/model/Utils/MapPolyline.java

```diff
@@ -58,24 +84,68 @@ public List<Point> getCoordinates(LineString lineString){
 
     /**
      * This method will plot the line on the map with a certain style and color.
-     * @param pointList list with coordinates.
-     * @param color polyline color.
+     * @param colors polyline color.
      */
-    public void drawPolyline(List<Point> pointList, int color) {
+    public void drawPolyline(List<List<Point>> points, List<Integer> colors){
         mapView.getMapAsync(mapboxMap -> {
             mapboxMap.getStyle(style -> {
-                LineString lineString = LineString.fromLngLats(pointList);
-                Feature feature = Feature.fromGeometry(lineString);
-                Source polylineSource = new GeoJsonSource("polyline-source", feature);
-                style.addSource(polylineSource);
-
-                LineLayer polylineLayer = new LineLayer("polyline-layer", "polyline-source");
-                polylineLayer.setProperties(
-                        PropertyFactory.lineColor(color),
-                        PropertyFactory.lineWidth(4f)
-                );
-                style.addLayer(polylineLayer);
+
+                List<Point> polyline1Points = points.get(0);
+                LineString polyline1 = LineString.fromLngLats(polyline1Points);
+
+                List<Point> polyline2Points = points.get(1);
+                LineString polyline2 = LineString.fromLngLats(polyline2Points);
+
+                List<Point> polyline3Points = points.get(2);
+                LineString polyline3 = LineString.fromLngLats(polyline3Points);
+
+
+                List<Feature> features = new ArrayList<>();
+                features.add(Feature.fromGeometry(polyline1));
+                features.add(Feature.fromGeometry(polyline2));
+                features.add(Feature.fromGeometry(polyline3));
+                FeatureCollection featureCollection = FeatureCollection.fromFeatures(features);
+
+                GeoJsonSource polyline1Source = new GeoJsonSource("polyline1-source", featureCollection);
+                GeoJsonSource polyline2Source = new GeoJsonSource("polyline2-source", featureCollection);
+                GeoJsonSource polyline3Source = new GeoJsonSource("polyline3-source", featureCollection);
+                style.addSource(polyline1Source);
+                style.addSource(polyline2Source);
+                style.addSource(polyline3Source);
+
+                if(polyline1Points != null){
```

Comment:
Find a way to refactor the if's and the method looks too big a recommendation is to use HashMap.